### DISCUSSION TITLE
Tailwindcss: Repeating classes

### DISCUSSION THREAD

**Discussion by stephan-buckmaster** [AllYourBot/hostedgpt#520] [General]
I got some feedback about a lot of copy/paste of the styling for llnks and button elements in the views of this app. How to improve this so it is maintainable? 

Tailwind has [a discussion about this](https://tailwindcss.com/docs/reusing-styles) leading up to the  `@apply` directive. 

What is a good way to organize this?

**Comment by keithschacht** [comment]
I really want to introduce view components. I’ve been spending a decent chunk of time digging into different ones: ViewComponent and Phlex being the two most popular, but I don’t love either of those. I have quite a list of ones I’ve been working through. I’d ideally like to lean on the new Web Components standard which is supported in all browsers, so I’m trying to find a good rails library that leans on the browser support. I need to do some more digging still.

I think components is a bit better of a solution than the @apply directive, although it’s going to be more complex to implement. However, it should result in a lot of nice wins.

Just for my own learning, where did this feedback come up?

**Reply by stephan-buckmaster** [reply]
Sounds good ... Came up in a discussion in the rubyonrails - link slack
